### PR TITLE
Add integration test CI jobs that use all the storages with the group commit feature enabled

### PR DESCRIPTION
**Pull request by komamitsu**
## Description

We already added the group commit feature for the coordinator table commit which contains an integration test CI job only using PostgreSQL 15. [We discussed how to do that before](https://github.com/scalar-labs/scalardb/pull/1728#discussion_r1611057235) and decided to add a daily/weekly schedule for the test. But, we noticed the longest duration of the integration tests with the group commit feature was about 16 minutes while the longest duration of the current normal integration tests is 24 minutes. So, in this PR we add integration test CI jobs for the group commit feature to the normal integration test cases that are run on `push` event and so on.

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/pull/1728

## Changes made

- Add CI jobs to `.github/workflows/ci.yaml` to run the consensus commit related integration test cases with the group commit feature
- Make `cassandara`, `cosmos`, `dynamo` and `multistorage` integration test Java classes accept and use the group commit related system properties passed by the GitHub Actions workflow
- Move some logic to handle the group commit related system properties from `ConsensusCommitJdbcEnv` to `ConsensusCommitUtils` so that other storages' integration tests can reuse it
- Change the system property names for the group commit integration tests from `scalar.db.xxx` to `scalardb.xxx` to align with [this change](https://github.com/scalar-labs/scalardb/commit/339203af6683ba68311cb5182a4bf6692ee6925c#diff-8661f4ba56dd608a0331a1b951fcc910c80ddd53de7dec30cdf6c0e773cab6d0R139) (I forgot to take care of that...)
- Prevent `integrationTestMultiStorage` from executing the JDBC integration tests that was introduced with [this change](https://github.com/scalar-labs/scalardb/commit/bb62b1af1e6dce1e49db12290155357b1a66f9d8#diff-8661f4ba56dd608a0331a1b951fcc910c80ddd53de7dec30cdf6c0e773cab6d0R59)

The last 2 changes are actually a bit unrelated bug fixes, but the fixes for them will be overlapped with this PR. So, the fixes are included in this PR, sorry.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

There is another option to add Java classes for the integration tests with the group commit feature like `ConsensusCommitIntegrationTestWithJdbcDatabaseWithGroupCommit` which would inherit `ConsensusCommitIntegrationTestWithJdbcDatabase`, not adding CI jobs. But the way increases the number of integration test cases per CI job and increase the total duration of CI. So, in this PR, we add additional CI jobs for that to run them in parallel.

## Release notes

N/A

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -20,6 +20,7 @@ sourceSets {
             compileClasspath += main.output + test.output
             runtimeClasspath += main.output + test.output
             srcDir file('src/integration-test/java')
+            include '**/com/scalar/db/common/*.java'
```
Comment (komamitsu) [inline core/build.gradle:23]: Only for newly added `ConsensusCommitTestUtils`

File: core/build.gradle [hunk 2]
```diff
@@ -57,7 +61,8 @@ sourceSets {
             compileClasspath += main.output + test.output
             runtimeClasspath += main.output + test.output
             srcDir file('src/integration-test/java')
-            include '**/com/scalar/db/storage/jdbc/*.java'
+            include '**/com/scalar/db/common/*.java'
+            include '**/com/scalar/db/storage/jdbc/JdbcTestUtils.java'
```
Comment (komamitsu) [inline core/build.gradle:65]: This is a bit unrelated bug fix to prevent this integration test task from executing JDBC integration tests.

File: core/src/integration-test/java/com/scalar/db/common/ConsensusCommitTestUtils.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import static com.scalar.db.transaction.consensuscommit.ConsensusCommitConfig.COORDINATOR_GROUP_COMMIT_METRICS_MONITOR_LOG_ENABLED;
+import static com.scalar.db.transaction.consensuscommit.ConsensusCommitConfig.COORDINATOR_GROUP_COMMIT_OLD_GROUP_ABORT_TIMEOUT_MILLIS;
+import static com.scalar.db.transaction.consensuscommit.ConsensusCommitConfig.COORDINATOR_GROUP_COMMIT_SLOT_CAPACITY;
+import static com.scalar.db.transaction.consensuscommit.ConsensusCommitConfig.COORDINATOR_GROUP_COMMIT_TIMEOUT_CHECK_INTERVAL_MILLIS;
+
+import java.util.Properties;
+
+public final class ConsensusCommitTestUtils {
+  private static final String PROP_CONSENSUS_COMMIT_COORDINATOR_GROUP_COMMIT_ENABLED =
+      "scalardb.consensus_commit.coordinator.group_commit.enabled";
```
Comment (komamitsu) [inline core/src/integration-test/java/com/scalar/db/common/ConsensusCommitTestUtils.java:15]: Renamed the prefix from `scalar.db.` to `scalardb.`

File: core/src/integration-test/java/com/scalar/db/storage/cassandra/ConsensusCommitCrossPartitionScanIntegrationTestWithCassandra.java [hunk 4]
```diff
@@ -13,7 +13,7 @@ public class ConsensusCommitCrossPartitionScanIntegrationTestWithCassandra
 
   @Override
   protected Properties getProps(String testName) {
-    Properties properties = CassandraEnv.getProperties(testName);
+    Properties properties = ConsensusCommitCassandraEnv.getProperties(testName);
```
Comment (komamitsu) [inline core/src/integration-test/java/com/scalar/db/storage/cassandra/ConsensusCommitCrossPartitionScanIntegrationTestWithCassandra.java:16]: These integration test classes need to use `ConsensusCommitCassandraEnv` to enable the group commit feature

**Comment by feeblefakie** [comment]
@komamitsu Thank you!
I might miss something, but I'm wondering if we need to test the group commit feature on various storage implementations.
Since it's happening on the transaction management layer, I don't think it's needed.
What do you think?

**Comment by komamitsu** [comment]
@feeblefakie Good point. Theoretically, it sounds reasonable. But, without actual tests, I think we can only say "the group commit feature should work with all the storages in theory, but we don't guarantee that since we don't regularly test all the storages" and it doesn't seem sufficient.

But, adding those tests consumes additional computation resource of GitHub Actions, and it might be a trade-off.

**Comment by feeblefakie** [comment]
@komamitsu 

> Good point. Theoretically, it sounds reasonable. But, without actual tests, I think we can only say "the group commit feature should work with all the storages in theory, but we don't guarantee that since we don't regularly test all the storages" and it doesn't seem sufficient.
>
> But, adding those tests consumes additional computation resource of GitHub Actions, and it might be a trade-off.

Thanks. Yes, of course, it's better to do, but my point is that running every test every time might cause too much resource consumption or waiting time. But it's not about the group commit specific thing, so let me talk with @brfrn169 about it.

@brfrn169 
Since we apply layered architecture, we should start thinking about what to test in every PR and what to test every day/week or something. Let's chat about it sometime soon.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by komamitsu** [comment]
@feeblefakie I see. In fact, I created [a branch](https://github.com/scalar-labs/scalardb/tree/daily-group-commit-it) for daily (or weekly) integration test for the group commit https://github.com/scalar-labs/scalardb/compare/339203af6683ba68311cb5182a4bf6692ee6925c..310c283b44c1df24de97ddd97a007c9406e9dc3a. Let me know if you think we'd better move the group commit related integration test to daily/weekly actions.

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by feeblefakie** [comment]
@komamitsu Can you check the conflict?

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!
